### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by aikanshagarwal**

### CODE REVIEW

File: src/main/java/com/example/springassignment3/configuration/Security.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception
+    {
+        auth.jdbcAuthentication().dataSource(dataSource).passwordEncoder(new BCryptPasswordEncoder());
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception
+    {
+        http.csrf().disable()
+                .authorizeRequests().antMatchers(HttpMethod.DELETE).hasRole("ADMIN").           //only admins are allowed to perform delete operations
```
Comment (tarunya882) [inline src/main/java/com/example/springassignment3/configuration/Security.java]: Use constants wherever required

File: src/main/java/com/example/springassignment3/service/PatientService.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+public interface PatientService
+{
+    public void savePatient(Patient thePatient);
+
+    public List<Patient> findAllPatients();
+
+    public Optional<Patient> findPatientById(int theId);
+
+    public void deletePatientById(int theId);
+
```
Comment (tarunya882) [inline src/main/java/com/example/springassignment3/service/PatientService.java:17]: Add method to update patient details

File: src/main/java/com/example/springassignment3/service/PatientServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+public class PatientServiceImpl implements PatientService
+{
+
+    @Autowired
+    private PatientRepository thePatientRepository;
+
+    @Override
+    public void savePatient(Patient thePatient)
+    {
+        thePatientRepository.save(thePatient);
```
Comment (tarunya882) [inline src/main/java/com/example/springassignment3/service/PatientServiceImpl.java:23]: Add logger statements

File: src/main/java/com/example/springassignment3/service/PatientServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    @Override
+    public List<Patient> findAllPatients()
+    {
+        return thePatientRepository.findAll();
+    }
+
+    @Override
+    public Optional<Patient> findPatientById(int theId)
+    {
+        return thePatientRepository.findById(theId);
```
Comment (tarunya882) [inline src/main/java/com/example/springassignment3/service/PatientServiceImpl.java:35]: Handle exception case when patient is not found

File: src/main/java/com/example/springassignment3/service/PatientServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @Override
+    public void deletePatientById(int theId)
+    {
+        thePatientRepository.deleteById(theId);
```
Comment (tarunya882) [inline src/main/java/com/example/springassignment3/service/PatientServiceImpl.java:42]: same here also. Handle exception cases wherever needed

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -1 +1,3 @@
-
+spring.datasource.url=jdbc:mysql://localhost:3306/patient_tracker
+spring.datasource.username=root
+spring.datasource.password=Family@5567
```
Comment (tarunya882) [inline src/main/resources/application.properties:3]: This looks fine

File: src/main/java/com/example/springassignment3/entity/Test.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    @Column(name = "name")
+    private String name;
+
+    @ManyToOne(cascade = {CascadeType.MERGE,CascadeType.DETACH,CascadeType.PERSIST,CascadeType.REFRESH})
+    @JoinColumn(name = "lab_id" , referencedColumnName = "id")
+    private Lab lab;
+
+    @ManyToMany(mappedBy = "tests")
+    private List<Patient> patients;
+
```
Comment (tarunya882) [inline src/main/java/com/example/springassignment3/entity/Test.java:32]: This looks good

File: src/main/java/com/example/springassignment3/configuration/Security.java (hunk 1 above)
Comment (aikanshagarwal) [reply src/main/java/com/example/springassignment3/configuration/Security.java]: Okay

File: src/main/java/com/example/springassignment3/service/PatientService.java (hunk 2 above)
Comment (aikanshagarwal) [reply src/main/java/com/example/springassignment3/service/PatientService.java:17]: Done

File: src/main/java/com/example/springassignment3/service/PatientServiceImpl.java (hunk 3 above)
Comment (aikanshagarwal) [reply src/main/java/com/example/springassignment3/service/PatientServiceImpl.java:23]: Yes , logger will be implemented

File: src/main/java/com/example/springassignment3/service/PatientServiceImpl.java (hunk 4 above)
Comment (aikanshagarwal) [reply src/main/java/com/example/springassignment3/service/PatientServiceImpl.java:35]: Exception handling will be taken care of

File: src/main/java/com/example/springassignment3/service/PatientServiceImpl.java (hunk 5 above)
Comment (aikanshagarwal) [reply src/main/java/com/example/springassignment3/service/PatientServiceImpl.java:42]: Sure

**Pull request merged**